CHANGELOG.md

```diff
@@ -1,3 +1,211 @@
+
+  - builds the driver provider in lib/driverProviders/index instead of lib/runner
+   closes #1945
+
+- ([681b54a](https://github.com/angular/protractor/commit/681b54a21ee1467d5a95c3693cde148759767d62))
+  refactor(browser): Remove protractor.wrapDriver() **breaking change** (#3827)
+
+  Before:
+
+  Users could create their own selenium driver instance and enable Protractor on
+  it like so:
+
+  ```js
+  let capabilities = webdriver.Capabilities.chrome();
+  let driver = new webdriver.Builder().usingServer(seleniumAddress)
+    .withCapabilities(capabilities).build();
+  let browser = protractor.wrapDriver(driver);
+  ```
+
+  Over the years, wrapDriver() has become increasingly broken as Protractor
+  needs extra configuration options that wrapDriver() doesn't set.
+
+  After:
+
+  This method is removed. If users need a new browser instance, they can
+  use `browser.forkNewDriverInstance()`.
+
+- ([86fd569](https://github.com/angular/protractor/commit/86fd56917f039efbff8e6f323f4d91fa8bc821a4))
+  feat(ngUpgrade): Auto detect ngUpgrade apps and make the ng12Hybrid flag
+  unnecessary for most users (#3847)
+
+
+## Bug fixes
+
+- ([de153e7](https://github.com/angular/protractor/commit/de153e769292f6b9a99b2d5152bd2929ab1c48af))
+  fix(launcher): running getMultiCapabilities should reject on errors (#3876)
+
+  closes #3875
+- ([1345137](https://github.com/angular/protractor/commit/1345137dc5173e868de4b9da6ed16b7928e4c50e))
+  fix(isElementPresent): for un-wrapped `WebElement`s, `browser.isElementPresent`
+  was broken (#3871)
+
+  Closes #3864
+- ([4af3b2e](https://github.com/angular/protractor/commit/4af3b2e30e925ea9d8e47537ea0a7fe8f04b579d))
+  fix(element): Fix typing of ElementFinder.then (#3835)
+
+  Type `then` as optional on ElementFinder.
+
+
+## Dependencies
+
+- ([7376708](https://github.com/angular/protractor/commit/7376708c723976ef8a0a3ad7c245606bef1221db))
+  deps(tslint): set tslint to ~4.2 (#3938)
+- ([cb38ed0](https://github.com/angular/protractor/commit/cb38ed0a8aae2cb862001e0b6f076aa9972f4489))
+  Refactor element explorer to work with selenium-webdriver 3 (#3828)
+
+  This implementation now relies mostly on promises explicitly, so the control
+  flow is only used to add one large task to the queue. This should pave the way
+  for the eventual removal of the control flow, as well as getting element
+  explorer to work immediately.
+
+  BREAKING CHANGE
+
+  You can no longer use the `repl` command from within `browser.pause()`. Instead,
+ use `browser.explore()` to directly enter the repl.
+- ([8196059](https://github.com/angular/protractor/commit/819605933d2dfef70b4332a727b3b3830e306817))
+  chore(dependency): switch to webdriver-manager 11.1.0 and remove
+  `--versions.chrome 2.26` from circle.yml (#3865)
+- ([397bf65](https://github.com/angular/protractor/commit/397bf65e088b640cf3612f9da678180f49939b84))
+  deps(update): move @types/jasmine to devDependencies (#3795)
```

Comment:
This one should get listed as a breaking change, and have the solution (add @types to your deps) clearly listed.